crates/uv-distribution-types/src/index.rs

```diff
@@ -82,6 +82,44 @@ pub struct Index {
     /// publish-url = "https://upload.pypi.org/legacy/"
     /// ```
     pub publish_url: Option<Url>,
+    /// The authentication policy for the index.
+    ///
+    /// All requests made to URLs with this index URL as a prefix will follow
+    /// the policy corresponding to this authentication policy.
```

Comment:
I think we'll need to expand on this quite a bit for it to be user-facing. It's not clear when or how to use this.